tests/DCPS/DynamicTypes/run_test.pl

```diff
@@ -15,18 +15,26 @@
 
 sub run_test {
   my @test_name_params = ("my_struct", "outer_struct", "inner_union", "outer_union");
+  my @test_extensibilities = ("_final", "_appendable");
+  my @xcdr_version_params = ("1", "2");
   foreach my $test_name_param(@test_name_params) {
-    my @reader_args = ("$test_name_param -DCPSConfigFile rtps_disc.ini -ORBLogFile recorder_$test_name_param.log -ORBDebugLevel 10 -DCPSDebugLevel 10");
-    my @writer_args = ("$test_name_param -DCPSConfigFile rtps_disc.ini -ORBLogFile publisher_$test_name_param.log -ORBDebugLevel 10 -DCPSDebugLevel 10");
-
-    $test->process("reader_$test_name_param", './Recorder/xtypes_dynamic_recorder', join(' ', @reader_args));
-    $test->start_process("reader_$test_name_param");
-
-    $test->process("writer_$test_name_param", './Pub/xtypes_dynamic_pub', join(' ', @writer_args));
-    $test->start_process("writer_$test_name_param");
-
-    $status |= $test->wait_kill("reader_$test_name_param", 15);
-    $status |= $test->wait_kill("writer_$test_name_param", 15);
+    foreach my $test_extensibility(@test_extensibilities) {
+      foreach my $xcdr_version_param(@xcdr_version_params) {
+        my @reader_args = ("$test_name_param" . "$test_extensibility $xcdr_version_param -DCPSConfigFile rtps_disc.ini -ORBLogFile recorder_$test_name_param" .
+                          "$test_extensibility" . "_XCDR$xcdr_version_param.log -ORBDebugLevel 10 -DCPSDebugLevel 10");
+        my @writer_args = ("$test_name_param" . "$test_extensibility $xcdr_version_param -DCPSConfigFile rtps_disc.ini -ORBLogFile publisher_$test_name_param" .
+                          "$test_extensibility" . "_XCDR$xcdr_version_param.log -ORBDebugLevel 10 -DCPSDebugLevel 10");
+
+        $test->process("reader_$test_name_param" . "$test_extensibility" . "_XCDR$xcdr_version_param", './Recorder/xtypes_dynamic_recorder', join(' ', @reader_args));
```

Comment:
You could pull out the names and make them variables so these lines don't have to be so long.

```suggestion
        my $reader_name = "reader_$test_name_param" . $test_extensibility . "_XCDR$xcdr_version_param";
        $test->process($reader_name, './Recorder/xtypes_dynamic_recorder', join(' ', @reader_args));
```

Could also use that name for the log file names.

Also `@reader_args` and `@writer_args` a treated like a arrays with multiple strings in them, hence the `join`, but there's only one string in them, so the array and the `join` aren't necessary.